js_src/src/index.js

```diff
@@ -0,0 +1,22 @@
+import * as rtl17 from './rtl-17.esm';
+import * as rtl18 from './_react-testing-library';
+
+// The entry point for the react_testing_library JS file.
+//
+// Conditionally exports, and populates the window with, a version of RTL that's
+// compatible with either React 17 or 18, depending on `window.React.version`.
+
+const reactVersion = window.React?.version;
```

Comment:
Since react is and external dependecy i think you could quite literally do like 
```
import React from 'react';

const reactVersion = React.version;
``` 
and it should under the hood convert that to a `window.React.version` call.